src/clr-angular/data/tree-view/tree.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, Input } from '@angular/core';
+import { TREE_FEATURES_PROVIDER, TreeFeaturesService } from './tree-features.service';
+
+@Component({
+  selector: 'clr-tree',
+  template: `
+    <ng-content></ng-content>
+  `,
+  providers: [TREE_FEATURES_PROVIDER],
```

Comment:
Is it useful to leave a comment here (and / or on line#51 of `tree-node.ts`) that is explicit in explaining how the service is instantiated and injected as a singlton only at the root of the tree (if declared) or at the root TreeNode?